### PR TITLE
feat(refactor): Move check_vcfparser_cli

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### This PR adds | fixes:

- See above

### How to test:

- Automatic and continuous test suite

### Expected outcome:
- [ ] Installation, unit and integration test suite pass

### Review:
- [ ] Code review
- [ ] Tests pass

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes. If applicable record manual test results in PR header
- [ ] **MINOR** - when you add functionality in a backwards compatible manner. If applicable record manual test results in PR header
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by ramprasadn — APPROVED** [review]
Looks good to me 👍

File: lib/MIP/Vcfparser.pm [hunk 1]
```diff
@@ -455,6 +455,93 @@ sub check_data_terms {
...
+            strict_type => 1,
+        },
+        select_outfile => { store => \$select_outfile, strict_type => 1, },
+    };
+
+    check( $tmpl, $arg_href, 1 ) or croak q{Could not parse arguments!};
+
+    use MIP::Log::MIP_log4perl qw{ retrieve_log };
+
+    my $log = retrieve_log( { log_name => $LOG_NAME, } );
```
Comment (jemten) [inline lib/MIP/Vcfparser.pm:504]:
```suggestion
    my $log = Log::Log4perl->get_logger($LOG_NAME);;
```

File: lib/MIP/Vcfparser.pm [hunk 2]
```diff
@@ -455,6 +455,93 @@ sub check_data_terms {
...
+            dependens_on_value => $range_feature_file,
+            msg =>
+q{Need to specify which feature column(s) to use with range feature file when annotating variants by using flag -rf_ac}
+              . $NEWLINE,
+            value => $range_feature_annotation_column,
+        },
+        select_feature_matching_column => {
+            dependens_on_value => $select_feature_file,
+            msg =>
+q{Need to specify which feature column to use with select feature file when selecting variants by using flag -sf_mc}
```
Comment (jemten) [inline lib/MIP/Vcfparser.pm:517]:
```suggestion
q{Need to specify which feature column to use with select feature file when selecting variants by using flag --select_feature_matching_column}
```

File: lib/MIP/Vcfparser.pm [hunk 3]
```diff
@@ -455,6 +455,93 @@ sub check_data_terms {
...
+
+    use MIP::Log::MIP_log4perl qw{ retrieve_log };
+
+    my $log = retrieve_log( { log_name => $LOG_NAME, } );
+
+    my %cli = (
+        range_feature_annotation_column => {
+            dependens_on_value => $range_feature_file,
+            msg =>
+q{Need to specify which feature column(s) to use with range feature file when annotating variants by using flag -rf_ac}
```
Comment (jemten) [inline lib/MIP/Vcfparser.pm:510]:
```suggestion
q{Need to specify which feature column(s) to use with range feature file when annotating variants by using flag ----range_feature_annotation_columns}
```

File: lib/MIP/Vcfparser.pm [hunk 4]
```diff
@@ -455,6 +455,93 @@ sub check_data_terms {
...
+            dependens_on_value => $select_feature_file,
+            msg =>
+q{Need to specify which feature column to use with select feature file when selecting variants by using flag -sf_mc}
+              . $NEWLINE,
+            value => $select_feature_matching_column,
+        },
+        select_outfile => {
+            dependens_on_value => $select_feature_file,
+            msg =>
+q{Need to specify a select outfile to use when selecting variants by using flag -sof}
```
Comment (jemten) [inline lib/MIP/Vcfparser.pm:524]:
```suggestion
q{Need to specify a select outfile to use when selecting variants by using flag --select_outfile}
```

**Review by jemten — APPROVED** [review]
Looks good. 
I think we moved away from the short option format some time ago 🤔

File: lib/MIP/Vcfparser.pm (hunk 2 above)
Comment (henrikstranneheim) [reply lib/MIP/Vcfparser.pm:517]: Good catch! I will fix

File: lib/MIP/Vcfparser.pm (hunk 1 above)
Comment (henrikstranneheim) [reply lib/MIP/Vcfparser.pm]: Done

**Comment by henrikstranneheim** [comment]
I will go ahead and merge this one as the failing tests are on pull and not for this PR. @ramprasadn I think this is an effect of your last PR to master (judging from the actions). Let's fix that in a separate PR

**Pull request merged**